packages/contracts/src/contracts/current/forwarder/Forwarder.sol

```diff
@@ -0,0 +1,48 @@
+pragma solidity ^0.4.24;
+pragma experimental ABIEncoderV2;
+
+import "./MixinForwarderCore.sol";
+import "./MixinBuyExactAssets.sol";
+import "./MixinMarketBuyTokens.sol";
+import "../utils/Ownable/Ownable.sol";
+
+contract Forwarder is
+    MixinForwarderCore,
+    MixinBuyExactAssets,
+    MixinMarketBuyTokens,
+    Ownable
+{
+    function Forwarder(
+        Exchange _exchange,
+        EtherToken _etherToken,
+        ZRXToken _zrxToken,
+        uint8 assetProxyId)
+        public
+        Ownable()
+    {
+        EXCHANGE = _exchange;
+        ETHER_TOKEN = _etherToken;
+        ZRX_TOKEN = _zrxToken;
+        setERC20ProxyApproval(assetProxyId);
```

Comment:
No since this is for moving ZRX and WETH, we receive the ERC721 (so the maker needs the allowance) then we send the ERC721 directly back to msg.sender.